.github/workflows/Tests.yaml

```diff
@@ -433,3 +434,129 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  arch_datastructures_tests:
+    name: DataStructures arch tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [gcc-9,
+                   clang-10
+                   ]
+        sde_arch: [-nhm,
+                   -snb,
+                   -hsw,
+                   -skl,
+                   -icx]
+        include:
+          - sde_arch: -nhm
+            arch_override: nehalem
+          - sde_arch: -snb
+            arch_override: sandybridge
+          - sde_arch: -hsw
+            arch_override: haswell
+          - sde_arch: -skl
+            arch_override: skylake
+          - sde_arch: -icx
+            arch_override: icelake-server
+          - sde_arch: -tgl
+            arch_override: tigerlake
+            compiler: clang-10
+        exclude:
+          - sde_arch: -icx
+            compiler: gcc-7
+          - sde_arch: -icx
+            compiler: clang-6.0
+    container:
+      image: moxcodes/spectrebuildenv:archtests
+      env:
+        # We run unit tests with the following compiler flags:
+        # - `-Werror`: Treat warnings as error.
+        # - `-march=x86-64`: Make sure we are building on a consistent
+        #   architecture so caching works. This is necessary because GitHub
+        #   may run the job on different hardware.
+        CXXFLAGS: "-Werror"
+        # We make sure to use a fixed absolute path for the ccache directory
+        CCACHE_DIR: /work/ccache
+        # GitHub Actions currently limits the size of individual caches
+        # to 400 MB.
+        CCACHE_MAXSIZE: 400M
+        CCACHE_COMPRESS: 1
+        CCACHE_COMPRESSLEVEL: 6
+        # We hash the content of the compiler rather than the location and mtime
+        # to make sure the cache works across the different machines
+        CCACHE_COMPILERCHECK: content
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Restore ccache
+        uses: actions/cache@v1
+        with:
+          path: /work/ccache
+          key:
+            "ccache-${{ matrix.compiler }}-Debug-pch-ON"
```

Comment:
Using ccache for these arch tests might not be worth the added complexity in this file. I assume compiling these tests is fast? Then I suggest you remove all ccache-related steps and variables from this job.

Reply:
Also, we have limited cache space on GitHub, which we should probably preferably use for the unit tests.